CHANGELOG.md

```diff
@@ -11,6 +11,13 @@ PLEASE REMEMBER TO CHANGE THE '..main' WITH AN ACTUAL TAG in GITHUB LINK.
 ## jax 0.2.29 (Unreleased)
 * [GitHub
   commits](https://github.com/google/jax/compare/jax-v0.2.28...main).
+  * In `jax.numpy` type promotion, values of type `np.uint64` now raise a `TypePromotionError`
```

Comment:
I think it's probably also worth emphasizing that it's the (uint64, int64)-> float promotion that is particularly odd. (uint64, float) -> float seems sensible to me, but it's the consequence of that in the lattice rules for uint64 and int64 that is troubling.

Reply:
Yeah, this is strange... Semantically it might be nice if `uint64,float -> float` were allowed, but `uint64,int -> float` were not. But I don't think that's possible within a lattice-based promotion approach.